src/Workleap.OpenApi.MSBuild/SpectralManager.cs

```diff
@@ -9,23 +8,45 @@ internal sealed class SpectralManager : ISpectralManager
 
     private readonly ILoggerWrapper _loggerWrapper;
     private readonly IHttpClientWrapper _httpClientWrapper;
-    private readonly string _toolDirectory;
-
-    public SpectralManager(LoggerWrapper loggerWrapper, string openApiToolsDirectoryPath, IHttpClientWrapper httpClientWrapper)
+    private readonly string _spectralDirectory;
+    private readonly string _openApiToolsDirectory;
+    private readonly IProcessWrapper _processWrapper;
+    
+    public SpectralManager(ILoggerWrapper loggerWrapper, string openApiToolsDirectoryPath, IHttpClientWrapper httpClientWrapper, IProcessWrapper processWrapper)
```

Comment:
General comment about exception / error handling:

Have we tested different scenarios where things goes wrong? I don't have a full list, but like:

- Failed HTTP download
- Invalid / malformed parameters (csproj properties, invalid yaml documents, invalid spectral ruleset url, invalid paths)

... and basically see how the whole thing reacts. Can a lambda developer understand what happened and address the issues without reaching out to our support channel?

Reply:
Yup that will be the focus for our next story. Error handling and proper upstream messaging